extensions/eclipse-che-theia-plugin-ext/src/browser/che-languages-test-api.ts

```diff
@@ -0,0 +1,177 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { ChePluginHandleRegistry } from './che-plugin-handle-registry';
+import { interfaces } from 'inversify';
+import { CheLanguagesTestAPI } from '../common/che-languages-test-protocol';
+import {
+    CompletionContext,
+    CompletionResultDto,
+    SignatureHelp,
+    Hover,
+    DocumentHighlight,
+    Range,
+    TextEdit,
+    FormattingOptions,
+    Definition,
+    DocumentLink,
+    CodeLensSymbol,
+    DocumentSymbol,
+    ReferenceContext,
+    Location,
+    SignatureHelpContext,
+    CodeActionContext,
+    CodeAction,
+    FoldingRange,
+} from '@theia/plugin-ext/lib/common/plugin-api-rpc-model';
+import { UriComponents } from '@theia/plugin-ext/lib/common/uri-components';
+import { CancellationToken, FoldingContext } from '@theia/plugin';
+import { SymbolInformation } from 'vscode-languageserver-types';
```

Comment:
is SymbolInformation should come from vscode-languageserver-types or it could come from `@theia/plugin` as well

Reply:
at the end language test api requests methods from LanguagesExt https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-api-rpc.ts#L1187 (for example see https://github.com/eclipse/che-theia/pull/802/files#diff-173df998f3e077e39aedf6ba7d61454fR58) and all these methods return types from `@theia/plugin-ext/lib/common/plugin-api-rpc-model` or `vscode-languageserver-types`  and they are different which we have in `@theia/plugin` that's why we can't use them

Reply:
yes for implementation I would say why not, I'm more concerned for the `che.d.ts`